hack/verify-staticcheck.sh

```diff
@@ -35,8 +35,9 @@ FOCUS="${FOCUS%/}" # Remove the ending "/"
 # See https://staticcheck.io/docs/checks
 CHECKS=(
   "all"
-  "-S1*"   # Omit code simplifications for now.
-  "-ST1*"  # Mostly stylistic, redundant w/ golint
+  "-S1*"    # Omit code simplifications for now.
+  "-ST1*"   # Mostly stylistic, redundant w/ golint
+  "-SA5011" # Possible nil pointer dereference
```

Comment:
not sure on this one, but we can always add new checks in the future.

Reply:
https://github.com/dominikh/go-tools/issues/656
It seems not correct in my opinion

Reply:
I think it's also reasonable to focus on ensuring we fix the existing checks throughout the project and keep the tool up to date over enabling every single check immediately anyhow.

Reply:
Agree.

But it may mislead new contributors to fix those that may not be a problem. We should check `-SA5011` again when upgrading static-check next time.  

To remove it now, we may add some guide information and comments in #92402.